plugins/search-react/src/components/SearchAutocomplete/SearchAutocomplete.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ChangeEvent, ReactNode, useCallback, useMemo } from 'react';
+
+import {
+  CircularProgress,
+  ListItemIcon,
+  ListItemText,
+  ListItemTextProps,
+} from '@material-ui/core';
+import {
+  Value,
+  Autocomplete,
+  AutocompleteProps,
+  AutocompleteChangeDetails,
+  AutocompleteChangeReason,
+  AutocompleteRenderInputParams,
+} from '@material-ui/lab';
+
+import { SearchContextProvider, useSearch } from '../../context';
+import { SearchBar, SearchBarProps } from '../SearchBar';
+
+/**
+ * Props for {@link SearchAutocomplete}.
+ *
+ * @public
+ */
+export type SearchAutocompleteProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined,
+> = Omit<
+  AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
+  'renderInput'
+> & {
+  inputDebounceTime?: SearchBarProps['debounceTime'];
+  renderInput?: (params: AutocompleteRenderInputParams) => JSX.Element;
+};
+
+const withContext = (
+  Component: <
+    T,
+    Multiple extends boolean | undefined,
+    DisableClearable extends boolean | undefined,
+    FreeSolo extends boolean | undefined,
+  >(
+    props: SearchAutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
+  ) => JSX.Element,
+) => {
+  return <
+    T,
+    Multiple extends boolean | undefined,
+    DisableClearable extends boolean | undefined,
+    FreeSolo extends boolean | undefined,
+  >(
+    props: SearchAutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
+  ) => (
+    <SearchContextProvider parentContext>
+      <Component {...props} />
+    </SearchContextProvider>
+  );
+};
+
+/**
+ * Recommended search autocomplete when you use the Search Provider or Search Context.
+ *
+ * @public
+ */
+export const SearchAutocomplete = withContext(
+  <
+    T,
+    Multiple extends boolean | undefined,
+    DisableClearable extends boolean | undefined,
+    FreeSolo extends boolean | undefined,
+  >({
+    loading,
+    value,
+    onChange = () => {},
+    options = [],
+    getOptionLabel = (option: T) => String(option),
+    renderInput,
+    inputDebounceTime,
+    fullWidth = true,
+    clearOnBlur = false,
+    ...rest
+  }: SearchAutocompleteProps<T, Multiple, DisableClearable, FreeSolo>) => {
+    const { setTerm } = useSearch();
```

Comment:
can we destruct the components props here instead? I think that will improve the readability of the api reports a bit.